src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerExpediteFinalOrder.java

```diff
@@ -152,27 +143,21 @@ public AboutToStartOrSubmitResponse<CaseData, State> aboutToSubmit(CaseDetails<C
         caseData.getFinalOrder().getExpeditedFinalOrderAuthorisation()
             .setExpeditedFinalOrderGeneralOrder(generalOrderToExpediteFinancialOrder.get().getValue());
 
-        generateFinalOrderCoverLetter.apply(details);
-        generateFinalOrder.apply(details);
+        documentGenerator.generateAndStoreCaseDocument(
```

Comment:
so this now only does the final order, not the final order coverletter. I assume that gets done inside the NotificationDispatcher.


Any reason not to use the existing case task?

Reply:
This will generically do any document type as long as there's a corresponding template content (which should already exist) so we're essentially replacing a class with a method call - imo that's better. Thoughts?

Reply:
yeah it's fine